event/buffered_bus.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package event
+
+import (
+	"sync/atomic"
+)
+
+type subscribeCommand *Subscription
+
+type unsubscribeCommand *Subscription
+
+type publishCommand Message
+
+type closeCommand struct{}
+
+// bufferedBus is a bus that uses a buffered channel to manage subscribers and publish messages.
+type bufferedBus struct {
```

Comment:
todo: Please add a compile time check to ensure (and document) that this type implements the `Bus` interface

Reply:
The constructor method does this already. I usually choose one or the other not both.

Reply:
Fair enough :)  Fred might ask you to change the constructor to return the concrete though :grin: